lib/ark_elixir/api.ex

```diff
@@ -1,19 +1,113 @@
 defmodule Ark_Elixir.Api do
     @moduledoc """
+    Operations for querying the API.
+    """
+
+    @doc """
+    Allow us to make a request on the network of our choice, using get_main or
+    get_dev, to the desired endpoint, with the possibility of passing optional
+    parameters and the matching headers.
+
+    The default network is Main.
+
+    ## Examples
 
+      iex> Ark_Elixir.Api.get("api/blocks") # Query on the main net
+      iex> Ark_Elixir.Api.get("api/blocks", [limit: 2]) # Query on the main net
+      iex> Ark_Elixir.Api.get("api/blocks", [limit: 2, network: "dev"]) # Query on the dev net
+      iex> Ark_Elixir.Api.get("api/blocks", [network: "dev", limit: 2]) # In any order
+      iex> Ark_Elixir.Api.get("api/blocks", [network: "dev"]) # No query params, just the network and the whole response
+      iex> Ark_Elixir.Api.get("api/blocks", [network: "bar"]) # Incorrect network, switching on the main one
+      iex> Ark_Elixir.Api.get("api/blocks", [network: :dev]) # You can specify the network as an Atom
     """
     def get(endpoint, opts \\ []) do
-        HTTPotion.get("https://api.arknode.net/#{endpoint}" <> query(opts),
-            headers: [nethash: "6e84d08bd299ed97c212c886c98a57e36545c8f5d645ca7eeae63a8bd62d8988", version: "1.0.1", port: "4001"])
-        |> Map.get(:body)
-        |> Poison.Parser.parse!
+      case opts[:network] do
+        "main" -> get_main(endpoint, opts)
+        :main -> get_main(endpoint, opts)
+        "dev" -> get_dev(endpoint, opts)
+        :dev -> get_dev(endpoint, opts)
+        "dark" -> get_dev(endpoint, opts)
+        :dark -> get_dev(endpoint, opts)
```

Comment:
I would pick either binary or atoms here and not support both. That way the API is more clear and easier to define with typespecs

Reply:
I see, I was thinking it may be a good idea but wasn't really sure about it. Is one choice better than the other ? And if yes, why ?

Reply:
If you are using them as constants atoms make more sense. Binaries are better for dynamic values specifically because of the fact that you can be DDoS'd by using dynamic atoms (since they aren't garbage collected)

In this case the only options are :dev and :main, which are constant, so it is fine to use atoms here.

Reply:
Rewriting it for atoms only then, good call !

Reply:
[Done](https://github.com/Highjhacker/Ark-Elixir/pull/3/commits/5b16cd572430b1736e9567e4d3736b1ebe159ebf).